src/iteration.rs

```diff
@@ -0,0 +1,176 @@
+use crate::{h3ron::H3Cell, node::Node};
+
+type NodeStackIter<'a, V> = std::iter::Flatten<std::slice::Iter<'a, Option<Box<Node<V>>>>>;
+
+pub(crate) struct Iter<'a, V> {
+    stack: Vec<NodeStackIter<'a, V>>,
+    #[allow(clippy::borrowed_box)]
+    curr: Option<&'a Box<Node<V>>>,
+}
```

Comment:
@maplant I'd love your opinion on this. I first tried to store a stack of slices which I'd manually resize as I walked the tree, but it was too damn confusing. It feels weird implementing an iterator with iterators. But if LLVM is doing its job, it should be fine.